File: Leetcode/cpp/sol3.cpp
// https://leetcode.com/problems/longest-substring-without-repeating-characters/description/

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        set<int> seen;
        int l = 0;
        int max_count = 0;
        for (int i = 0; i < s.size(); i++){
            while(seen.count(s[i])){
                seen.erase(s[l]);
                l++;
            }
            seen.insert(s[i]);
            int current = i - l+1;
            cout << current << endl;
            // max_count = max_count ? max_count < current: current;
            max_count = max(max_count, current);
        }
        return max_count;
    }
};

File: Leetcode/python/sol3.py
# https://leetcode.com/problems/longest-substring-without-repeating-characters/
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        unique :set = set()
        l :int = 0
        max_count :int = 0
        for i in range(len(s)):
            while s[i] in unique:
                unique.remove(s[l])
                l+=1
            unique.add(s[i])
            max_count = max(max_count, i - l +1)

        return max_count




